protocols/messages.go

```diff
@@ -24,13 +25,13 @@ type PayloadType string
 
 // CreateObjectivePayload generates an objective message from the given objective id and payload.
 // CreateObjectivePayload handles serializing `p` into json.
-func CreateObjectivePayload(id ObjectiveId, payloadType PayloadType, p interface{}) ObjectivePayload {
+func CreateObjectivePayload(id ObjectiveId, payloadType PayloadType, p interface{}) (ObjectivePayload, error) {
 	b, err := json.Marshal(p)
 	if err != nil {
-		panic(err)
+		return ObjectivePayload{}, err
```

Comment:
```suggestion
		return ObjectivePayload{}, fmt.Errorf("failed to marshal payload: %w", err)
```

Maybe? (fmt not already imported)